src/mumble/MainWindow.cpp

```diff
@@ -1272,37 +1272,91 @@ void MainWindow::openUrl(const QUrl &url) {
  * directly after server synchronization is completed.
  * @see void MainWindow::msgServerSync(const MumbleProto::ServerSync &msg)
  */
-void MainWindow::findDesiredChannel() {
-	bool found          = false;
-	QStringList qlChans = qsDesiredChannel.split(QLatin1String("/"));
-	Channel *chan       = Channel::get(Mumble::ROOT_CHANNEL_ID);
-	QString str         = QString();
-	while (chan && qlChans.count() > 0) {
-		QString elem = qlChans.takeFirst().toLower();
-		if (elem.isEmpty())
-			continue;
-		if (str.isNull())
-			str = elem;
-		else
-			str = str + QLatin1String("/") + elem;
-		foreach (Channel *c, chan->qlChannels) {
-			if (c->qsName.toLower() == str) {
-				str   = QString();
-				found = true;
-				chan  = c;
-				break;
-			}
-		}
-	}
-	if (found) {
-		if (chan != ClientUser::get(Global::get().uiSession)->cChannel) {
-			Global::get().sh->joinChannel(Global::get().uiSession, chan->iId);
-		}
-		qtvUsers->setCurrentIndex(pmModel->index(chan));
-	} else if (Global::get().uiSession) {
-		qtvUsers->setCurrentIndex(pmModel->index(ClientUser::get(Global::get().uiSession)->cChannel));
-	}
-	updateMenuPermissions();
+ void MainWindow::findDesiredChannel() {
+    QStringList qlChans = qsDesiredChannel.split(QLatin1String("/"));
+    Channel *chan = Channel::get(Mumble::ROOT_CHANNEL_ID);
+    QString str = QString();
+    
+    // Try case-sensitive search first (existing behavior)
+    bool found = findChannelWithSensitivity(chan, qlChans, str, true);
+    
+    // If not found, try case-insensitive search
+    if (!found) {
+        // Reset for case-insensitive search
+        chan = Channel::get(Mumble::ROOT_CHANNEL_ID);
+        qlChans = qsDesiredChannel.split(QLatin1String("/"));
+        str = QString();
+        
+        found = findChannelWithSensitivity(chan, qlChans, str, false);
+    }
+    
+    if (found) {
+        if (chan != ClientUser::get(Global::get().uiSession)->cChannel) {
+            Global::get().sh->joinChannel(Global::get().uiSession, chan->iId);
+        }
+        qtvUsers->setCurrentIndex(pmModel->index(chan));
+    } else if (Global::get().uiSession) {
+        qtvUsers->setCurrentIndex(pmModel->index(ClientUser::get(Global::get().uiSession)->cChannel));
+    }
+    
+    updateMenuPermissions();
+}
+
+// Helper method to find channel with either case-sensitive or case-insensitive search
+bool MainWindow::findChannelWithSensitivity(Channel *&chan, QStringList &qlChans, QString &str, bool caseSensitive) {
```

Comment:
Delegating the job to a separate function is perfectly fine and appreciated, however I would change the name to something more specific such as:

```
findChannelWithCasing()
```